src/lib/AST.cpp

```diff
@@ -1,30 +1,73 @@
 #include "AST.h"
 
 
-// Stores single number in Leaf
-TreeLeaf::TreeLeaf(double val) {
-    value = val;
+// Evaluates equality between values
+bool variableVal::operator == (const variableVal& rVal) const{ 
+    if(type != rVal.type){
+        return false;
+    }
+
+    switch(type){
+        case ReturnType::NONE:
+            throw std::runtime_error("Compared NONE type, programmed wrong");
+        case ReturnType::NUL:
+            return true;
+        case ReturnType::NUM:
```

Comment:
should we check if both right and left children are the same type - to prevent memory errors / incorrect evaluations?